### PR TITLE
Loki Queries: Query Splitting enabled by default

### LINKED ISSUE #67547: Loki: Make query splitting generally available
This issue is to track rolling out query splitting to the widest possible audience, and clearing any blockers therein. Some notes [here](https://docs.google.com/document/d/1qm2k_O7MWTiIqAc-jNYAYDhaAd8r22XKY0XsFTnnE64/edit#heading=h.2567nn2d4qkf).

NOTE: we should also re-evaluate whether the custom-timeout-support for logs-volume-histogram is still needed, considering query-splitting should help a lot there. ( i mean the https://github.com/grafana/grafana/blob/fcb14d2548d68dadb981f24dd76489088dd232d7/docs/sources/explore/logs-integration.md#logs-visualization document, where it deals with the `X-Query-Tags` header)

### PR DESCRIPTION
**Pull request by matyax**
In response to different query performance scenarios, we implemented query splitting, where queries that request more than a day of data are split into sub-requests of 1 day duration each. For example, requesting 7 days of logs will produce 7 requests of 1 day.

This PR makes this the default behavior for querying Loki.

## How it works

Any metric or log request that is made from Grafana in Dashboards or Explore that uses a time interval longer than one day will be split in one-day intervals, each one resulting in an individual and sequential request, from the most recent to the oldest. For example, if the user sends a metric request for 7 days of data, we generate request for the following sequence: day 7, day 6, day 5, ..., day 1.

## Can it be disabled

If needed, this feature can be disabled by setting `lokiQuerySplitting` to `false`. 

**Which issue(s) does this PR fix?**:

Closes https://github.com/grafana/grafana/issues/67547

**Special notes for your reviewer:**

Remove the feature flag from custom.ini.
The default querying behavior should be with query splitting for queries with > 1 day duration.

### CODE REVIEW

**Review by ivanahuckova — COMMENTED** [review]
@matyax you mentioned that you created a dashboard with couple of different panels to test query splitting + transformations. Would you commit it in this PR into `devenv/dev-dashboards/datasource-loki/loki_fakedata.json` to make it easier (now and in the future) test possible issues related to splitting + transformations/visualisations.

**Comment by matyax** [comment]
That is a very very good idea. I will do it 👍

**Comment by matyax** [comment]
Added ✅

File: pkg/services/store/testdata/public_testdata.golden.jsonc [hunk 1]
```diff
@@ -12,14 +12,14 @@
 //  }
 //  Name: 
 //  Dimensions: 3 Fields by 2 Rows
-//  +--------------------+----------------------+---------------+
```
Comment (ivanahuckova) [inline pkg/services/store/testdata/public_testdata.golden.jsonc:15]: @matyax do you know what is this?

File: devenv/dev-dashboards/datasource-loki/loki_query_splitting.json [hunk 2]
```diff
@@ -0,0 +1,900 @@
...
+    "templating": {
+      "list": []
+    },
+    "time": {
+      "from": "now-6h",
+      "to": "now"
+    },
+    "timepicker": {},
+    "timezone": "",
+    "title": "Loki Query splitting",
```
Comment (ivanahuckova) [inline devenv/dev-dashboards/datasource-loki/loki_query_splitting.json:896]:
```suggestion
    "title": "Datasource tests - Loki query splitting",
```

Based on convention for data sources
image

**Review by ivanahuckova — COMMENTED** [review]
Left a comment about naming of dashboard. Plus there are some changes in  `pkg/services/store/testdata/public_testdata.golden.jsonc` that I am not sure should be there. Otherwise LGTM.

File: pkg/services/store/testdata/public_testdata.golden.jsonc (hunk 1 above)
Comment (matyax) [reply pkg/services/store/testdata/public_testdata.golden.jsonc:15]: The build was failing, ran the Go tests and this change was made. What is this file? I have no idea.

Comment (gtk-grafana) [reply pkg/services/store/testdata/public_testdata.golden.jsonc:15]: Try merging in latest main and reverting this change out and running the tests again, it should go away?

Comment (matyax) [reply pkg/services/store/testdata/public_testdata.golden.jsonc:15]: 👌  will try.

**Comment by matyax** [comment]
Synced with main, removed the unwanted change.

**Review by ivanahuckova — APPROVED** [review]
🚀

**Pull request merged**